contracts/smart-contract-wallet/SmartAccount.sol

```diff
@@ -312,30 +325,26 @@ contract SmartAccount is
         uint8 v;
         bytes32 r;
         bytes32 s;
-        uint256 i = 0;
         address _signer;
-        (v, r, s) = signatureSplit(signatures, i);
+        (v, r, s) = signatureSplit(signatures);
         //todo add the test case for contract signature
         if(v == 0) {
             // If v is 0 then it is a contract signature
             // When handling contract signatures the address of the signer contract is encoded into r
             _signer = address(uint160(uint256(r)));
 
             // Check that signature data pointer (s) is not pointing inside the static part of the signatures bytes
-                // This check is not completely accurate, since it is possible that more signatures than the threshold are send.
-                // Here we only check that the pointer is not pointing inside the part that is being processed
-                require(uint256(s) >= uint256(1) * 65, "BSA021");
-
-                // Check that signature data pointer (s) is in bounds (points to the length of data -> 32 bytes)
-                require(uint256(s) + 32 <= signatures.length, "BSA022");
+                // Here we check that the pointer is not pointing inside the part that is being processed
+                if(uint256(s) < 65) revert WrongContractSignatureFormat(uint256(s), 0, 0);
```

Comment:
taking a note so it helps with resolving conflicts

Reply:
with using oz library we would always know that signature.length should be 65. I think i have added one check regarding this in QA.
should above be <= 65 ?

Reply:
it was require(uint256(s) >= 65) , so I just made an opposite, not including 65

Reply:
ah i see, my bad